Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/callback/openSAML/OpenSAML2ComponentBuilder.java

```diff
@@ -909,29 +690,26 @@ Attribute createHomeCommunityAttribute(final String communityId) {
      * @return the signature
      * @throws SAMLComponentBuilderException the exception
      */
-    public Signature createSignature(final X509Certificate certificate, final PrivateKey privateKey, final PublicKey publicKey)
-            throws SAMLAssertionBuilderException {
+    public Signature createSignature(final X509Certificate certificate, final PrivateKey privateKey,
+        final PublicKey publicKey) throws SAMLAssertionBuilderException {
         final BasicX509Credential credential = new BasicX509Credential(certificate, privateKey);
         credential.setEntityCertificate(certificate);
         credential.setPrivateKey(privateKey);
 
-
-        // credential.setPublicKey(publicKey);
         final Signature signature = OpenSAMLUtil.buildSignature();
         signature.setSigningCredential(credential);
 
         signature.setSignatureAlgorithm(SignatureConstants.ALGO_ID_SIGNATURE_RSA_SHA1);
 
         signature.setCanonicalizationAlgorithm(SignatureConstants.ALGO_ID_C14N_EXCL_OMIT_COMMENTS);
         try {
-            signature.setKeyInfo(getKeyInfo(certificate, publicKey));
-        } catch (Exception ex) {
-        	LOG.error("Get HL7 Prefix Property exception: {}", ex.getLocalizedMessage(), ex);
-        	throw new SAMLAssertionBuilderException(ex.getLocalizedMessage(), ex);
+            signature.setKeyInfo(getKeyInfo(publicKey));
+        } catch (final Exception ex) {
+            LOG.error("Get HL7 Prefix Property exception: {}", ex.getLocalizedMessage(), ex);
```

Comment:
Consider not logging the full exception here, since it's being passed up the stack, to reduce log verbosity.